gcloud-java-compute/src/main/java/com/google/gcloud/compute/InstanceInfo.java

```diff
@@ -0,0 +1,1282 @@
+            .setKey(entry.getKey()).setValue(entry.getValue()));
+      }
+      metadataPb.setItems(itemsPb);
+      metadataPb.setFingerprint(fingerprint);
+      return metadataPb;
+    }
+
+    /**
+     * Returns a builder for a {@code Metadata} object.
+     */
+    public static Builder builder() {
+      return new Builder();
+    }
+
+    /**
+     * Returns a {@code Metadata} object given the the metadata as a map. The total size of all keys
+     * and values must be less than 512 KB. Keys must conform to the following regexp:
+     * {@code [a-zA-Z0-9-_]+}, and be less than 128 bytes in length. This is reflected as part of a
+     * URL in the metadata server. Additionally, to avoid ambiguity, keys must not conflict with any
+     * other metadata keys for the project. Values must be less than or equal to 32768 bytes.
+     */
+    public static Metadata of(Map<String, String> values) {
+      return builder().values(values).build();
+    }
+
+    static Metadata fromPb(com.google.api.services.compute.model.Metadata metadataPb) {
+      Builder builder = builder();
+      if (metadataPb.getItems() != null) {
+        Map<String, String> metadataValues =
+            Maps.newHashMapWithExpectedSize(metadataPb.getItems().size());
+        for (com.google.api.services.compute.model.Metadata.Items item : metadataPb.getItems()) {
+          metadataValues.put(item.getKey(), item.getValue());
+        }
+        builder.values(metadataValues);
+      }
+      return builder.fingerprint(metadataPb.getFingerprint()).build();
+    }
+  }
+
+  /**
+   * A service account, with its specified scopes, authorized for this instance.
+   *
+   * @see <a href="https://cloud.google.com/compute/docs/authentication">Authenticating from Google
+   *     Compute Engine</a>
+   */
+  public static final class ServiceAccount implements Serializable {
+
+    static final Function<com.google.api.services.compute.model.ServiceAccount, ServiceAccount>
+        FROM_PB_FUNCTION =
+        new Function<com.google.api.services.compute.model.ServiceAccount, ServiceAccount>() {
+          @Override
+          public ServiceAccount apply(com.google.api.services.compute.model.ServiceAccount pb) {
+            return ServiceAccount.fromPb(pb);
+          }
+        };
+    static final Function<ServiceAccount, com.google.api.services.compute.model.ServiceAccount>
+        TO_PB_FUNCTION =
+        new Function<ServiceAccount, com.google.api.services.compute.model.ServiceAccount>() {
+          @Override
+          public com.google.api.services.compute.model.ServiceAccount apply(
+              ServiceAccount metadata) {
+            return metadata.toPb();
+          }
+        };
+
+    private static final long serialVersionUID = 4199610694227857331L;
+
+    private final String email;
+    private final List<String> scopes;
+
+    private ServiceAccount(String email, List<String> scopes) {
```

Comment:
In principle, yes. But we haven't been needing a `ServiceAccount` object yet. I'll open an issue to keep track of this.